Sources/HiPMobile/ServiceAccessLayer/ContentApiAccesses/UserRatingApiAccess.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (C) 2017 History in Paderborn App - Universität Paderborn
+//  
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//  
+//      http://www.apache.org/licenses/LICENSE-2.0
+//  
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.ServiceAccessLayer.ContentApiAccesses.Contracts;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.ServiceAccessLayer.ContentApiDtos;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Models;
+using System;
+using System.Net;
+
+namespace PaderbornUniversity.SILab.Hip.Mobile.Shared.ServiceAccessLayer.ContentApiAccesses
+{
+    public class UserRatingApiAccess : IUserRatingApiAccess
+    {
+
+        private readonly IContentApiClient contentApiClient;
+
+        private const string RequestPath = "/Exhibits/Rating/";
+
+        public UserRatingApiAccess(IContentApiClient contentApiClient)
+        {
+            this.contentApiClient = contentApiClient;
+        }
+
+        public async Task<UserRatingDto> GetUserRating(Exhibit exhibit)
+        {
+            try
+            {
+                string url = RequestPath + exhibit.IdForRestApi;
+                var json = await contentApiClient.GetResponseFromUrlAsString(url);
+                return JsonConvert.DeserializeObject<UserRatingDto>(json);
+            }
+            catch (NotFoundException)
+            {
+                return new UserRatingDto();
```

Comment:
Is this a good idea? Is this a situation that can actually legally happen?

Reply:
Yes. If the server is not reachable or something with the link is wrong, a NotFoundException will be thrown and a unknown user rating is shown. It's also possible to return null and set the user rating to unknown somewhere else, e.g. in the user rating manager

Reply:
If the server is not reachable, a `NotFoundException` shouldn't be thrown actually, if my memory serves me right. And if something with the link is wrong, then it makes sense to fail here instead of silently returning data the API actually never sent. `ApiAccess` classes should be dumb and simply act as an interface to the server, at least in my opinion. What do you think?

Reply:
A NotFoundException is thrown when their is no user rating for the exhibit id on the server. 
For the moment I have removed the try catch block because of each exhibit a user rating should be existing, but it would be the best if we talk about this issue on monday.

Reply:
Yes, let's discuss on Monday.